src/docs/patterns/appbar.mdx

```diff
@@ -107,27 +110,30 @@ The collapsible appbar should transition smoothly between the expanded and colla
 
 A very common pattern seen in many types of applications is the ability to search for data on a page. This is achieved by placing a search icon/button in the app bar. When this button is clicked, a white search bar slides in from the right to cover the default app bar. As the user enters text into the search field, the data is filtered in real-time. A back icon will close the Search Bar and clear the search results.
 
-If your search is not a global-level search, you can place a smaller search input near the content being searched (e.g., a card header).
-
-<React.Fragment>
-    <ImageGrid
-        regularWidth
-        images={[MiniSearch]}
-        caption={'Use a contextual search input if your search functionality is not at the page level.'}
-    />
-</React.Fragment>
-
 ## Auto-complete
 
 For more advanced searching, you may wish to include an auto-complete feature. You can implement autocomplete by including a contextual menu below the search bar with suggested search items. Alternatively, you can use type-ahead to automatically update the recommended search query in the search field as the user types (Google search does this).
 
+<ImageGrid
+    regularWidth
+    images={[AutoVariants]}
+    caption={[
+        '1. Search suggestions are provided in a dropdown menu.',
+        '2. The most likely search is shown in the search field as the user types.',
+    ]}
+/>
+
+## Contextual search
+
+If your search is not a global-level search, you can place your search input in different UI hierarchies to indicate the search context.
+
 <React.Fragment>
     <ImageGrid
-        regularWidth
-        images={[AutoVariants]}
+        images={[SiteSearch, PageSearch, ComponentSearch]}
         caption={[
-            '1. Search suggestions are provided in a dropdown menu.',
-            '2. The most likely search is shown in the search field as the user types.',
+            '1: Site-wide search, a.k.a. global search, is always placed in the primary app bar.',
+            '2: Page-wide search, usually placed in a secondary toolbar.',
```

Comment:
The second one feels strange to me. Are we steering people to do double appbars like that? What if you have a sitewide search and a page-level search? What about for Mobile?

Might want to call out some of these cases.

Reply:
It was more like an effort to distinguish actions at different hierarchies. So if you place something in a primary toolbar, it is assumed that this will be a global action. If you place icons in a secondary toolbar, then it applies to that page only. If you put it on a component bar, then it pertains to that component.

This will be better suited to a behavior section on its own. But that requires some extra effort so I did not do that...

Reply:
I think secondary toolbar for search just feels weird to me. I would assume that global or page-level search would go in the main toolbar and it would just depend on context which one it was (I wouldn't expect many applications to a global search **and** page-level search at the same time). Adding a secondary toolbar just to do page-level search seems overkill when you have a perfectly good primary toolbar. Other patterns (appbar) describe using the main toolbar for page-level actions as well.